### PR TITLE
Fix a few data types in the config docs

### PR DESCRIPTION
**Pull request by chrisbygrave**
- fix a few data types that aren't correct in config.md
- update contributors URL - dead link to the (I believe) correct link
- gas oracle modes - `fixed` is referenced in a few examples but wasn't included in the list of supported values

Signed-off-by: Chris Bygrave <chris.bygrave@gmail.com>

### CODE REVIEW

File: config.md [hunk 1]
```diff
@@ -91,7 +91,7 @@
 |---|-----------|----|-------------|
 |count|The maximum number of times to retry|`int`|`5`
 |enabled|Enables retries|`boolean`|`false`
-|factor|The retry backoff factor|`boolean`|`2`
+|factor|The retry backoff factor|`int`|`2`
```
Comment (peterbroadhurst) [inline config.md:94]: I'm actually confused by this one. This should be picking up the global config, all the way up from here - but that doesn't seem to have the bug 🤔 
https://github.com/hyperledger/firefly-common/blob/5a2539a3727b6eeecee07da0297198f7bcbd2bed/pkg/i18n/en_base_config_descriptions.go#L44

File: config.md [hunk 2]
```diff
@@ -137,7 +137,7 @@
 
 |Key|Description|Type|Default Value|
 |---|-----------|----|-------------|
-|factor|Factor to increase the delay by, between each retry|`boolean`|`2`
+|factor|Factor to increase the delay by, between each retry|`int`|`2`
```
Comment (peterbroadhurst) [inline config.md:140]: https://github.com/hyperledger/firefly-transaction-manager/blob/f319338146cf44c39695497f8f3f17a9169225d0/internal/tmmsgs/en_config_descriptions.go#L74

File: config.md [hunk 3]
```diff
@@ -104,10 +104,10 @@
 |credentials|CORS setting to control whether a browser allows credentials to be sent to this API|`boolean`|`true`
 |debug|Whether debug is enabled for the CORS implementation|`boolean`|`false`
 |enabled|Whether CORS is enabled|`boolean`|`true`
-|headers|CORS setting to control the allowed headers|`string`|`[*]`
+|headers|CORS setting to control the allowed headers|`[]string`|`[*]`
```
Comment (peterbroadhurst) [inline config.md:107]: This one is here: https://github.com/hyperledger/firefly-common/blob/5a2539a3727b6eeecee07da0297198f7bcbd2bed/pkg/i18n/en_base_config_descriptions.go#L104

File: config.md [hunk 4]
```diff
@@ -104,10 +104,10 @@
...
 |maxAge|The maximum age a browser should rely on CORS checks|[`time.Duration`](https://pkg.go.dev/time#Duration)|`600`
-|methods| CORS setting to control the allowed methods|`string`|`[GET POST PUT PATCH DELETE]`
-|origins|CORS setting to control the allowed origins|`string`|`[*]`
+|methods| CORS setting to control the allowed methods|`[]string`|`[GET POST PUT PATCH DELETE]`
```
Comment (peterbroadhurst) [inline config.md:109]: This one is here: https://github.com/hyperledger/firefly-common/blob/5a2539a3727b6eeecee07da0297198f7bcbd2bed/pkg/i18n/en_base_config_descriptions.go#L106

File: config.md [hunk 5]
```diff
@@ -104,10 +104,10 @@
...
+|origins|CORS setting to control the allowed origins|`[]string`|`[*]`
```
Comment (peterbroadhurst) [inline config.md:110]: This one is here: https://github.com/hyperledger/firefly-common/blob/5a2539a3727b6eeecee07da0297198f7bcbd2bed/pkg/i18n/en_base_config_descriptions.go#L107

File: config.md [hunk 6]
```diff
@@ -210,7 +210,7 @@
 |idleTimeout|The max duration to hold a HTTP keepalive connection between calls|[`time.Duration`](https://pkg.go.dev/time#Duration)|`475ms`
 |maxIdleConns|The max number of idle connections to hold pooled|`int`|`100`
 |method|The HTTP Method to use when invoking the Gas Oracle REST API|`string`|`GET`
-|mode|The gas oracle mode|connector | restapi | disabled|`connector`
+|mode|The gas oracle mode - connector, restapi, fixed, or disabled|`string`|`connector`
```
Comment (peterbroadhurst) [inline config.md:213]: This one is here: https://github.com/hyperledger/firefly-transaction-manager/blob/f319338146cf44c39695497f8f3f17a9169225d0/internal/tmmsgs/en_config_descriptions.go#L57

File: config.md [hunk 7]
```diff
@@ -249,7 +249,7 @@
 
 |Key|Description|Type|Default Value|
 |---|-----------|----|-------------|
-|factor|The retry backoff factor|`boolean`|`2`
+|factor|The retry backoff factor|`int`|`2`
```
Comment (peterbroadhurst) [inline config.md:252]: This will be fixed by the same `firefly-common` fix, once worked out.

**Review by peterbroadhurst — CHANGES_REQUESTED** [review]
Hi @chrisbygrave, this is great!
However... these types are actually built into Markdown from source, and the build checks what's checked in is matched with the latest generation.
You have to run `make reference` before committing, otherwise the `make` build fails showing you the mismatch.

So they need to be changed where the parameters are defined - which in this case is actually in some pre-reqs.
I've put in some pointers, and there's one that's confused me as to how we're getting the bug you point out.

**Comment by chrisbygrave** [comment]
Thanks for the pointers @peterbroadhurst! I hadn't spotted this was generated - I'll work through the pointers in a couple of follow-on PRs

File: config.md (hunk 1 above)
Comment (chrisbygrave) [reply config.md:94]: [Here's the culprit](https://github.com/hyperledger/firefly-common/blob/5a2539a3727b6eeecee07da0297198f7bcbd2bed/pkg/i18n/en_base_config_descriptions.go#L30)

```
var FloatType = "`boolean`"
```

Comment (peterbroadhurst) [reply config.md:94]: 🤦

File: config.md (hunk 6 above)
Comment (chrisbygrave) [reply config.md]: https://github.com/hyperledger/firefly-transaction-manager/pull/39

File: config.md (hunk 3 above)
Comment (chrisbygrave) [reply config.md]: https://github.com/hyperledger/firefly-common/pull/36

File: config.md (hunk 2 above)
Comment (chrisbygrave) [reply config.md]: https://github.com/hyperledger/firefly-common/pull/36

File: config.md (hunk 1 above)
Comment (chrisbygrave) [reply config.md]: https://github.com/hyperledger/firefly-common/pull/36

File: config.md (hunk 7 above)
Comment (chrisbygrave) [reply config.md]: https://github.com/hyperledger/firefly-common/pull/36

**Comment by chrisbygrave** [comment]
Ok, so:

- this PR simply fixes the URL in the CONTRIBUTING.md
- https://github.com/hyperledger/firefly-transaction-manager/pull/39 fixes the gas oracle mode
- https://github.com/hyperledger/firefly-common/pull/36 fixes the factor data type, and also the string array data types

Searching across hyperledger, I didn't find anywhere else using the `FloatType` other than firefly-transaction-manager, which uses it as the type for the event streams retry factor - another place we wanted to fix.

**Comment by peterbroadhurst** [comment]
ok - release up for common here, that'll need to go into the FFTM PR

**Comment by chrisbygrave** [comment]
I'll update this one once accordingly we get the FFTM PR through

**Comment by peterbroadhurst** [comment]
https://github.com/hyperledger/firefly-transaction-manager/releases/tag/v1.1.1 ready to pull in here now @chrisbygrave

**Review by peterbroadhurst — APPROVED** [review]

**Comment by peterbroadhurst** [comment]
Thanks @chrisbygrave for working this through all the layers 💪

**Pull request merged**